src/hotspot/share/gc/shared/gcLocker.cpp

```diff
@@ -33,11 +33,35 @@
 #include "runtime/javaThread.inline.hpp"
 #include "runtime/safepoint.hpp"
 #include "runtime/threadSMR.hpp"
+#include "utilities/ticks.hpp"
 
 volatile jint GCLocker::_jni_lock_count = 0;
 volatile bool GCLocker::_needs_gc       = false;
 unsigned int GCLocker::_total_collections = 0;
 
+// GCLockerTimingDebugLogger tracks specific timing information for GC lock waits.
+class GCLockerTimingDebugLogger : public StackObj {
+  private:
+    const char*  _log_message;
+    Ticks _start;
+
+  public:
+    GCLockerTimingDebugLogger(const char* log_message) :
+      _log_message(log_message) {
+        assert(_log_message != nullptr, "GC locker debug message must be set.");
+        _start = Ticks::now();
+      }
+
+    ~GCLockerTimingDebugLogger() {
+      const Tickspan elapsed_time = Ticks::now() - _start;
```

Comment:
Why is this outside the `if` logger-enabled check?

Reply:
Please move within the `if`.